ios/PurchasesHybridCommon/PurchasesHybridCommonTests/StoreProduct+HybridAdditionsTests.swift

```diff
@@ -0,0 +1,191 @@
+            }
+            it("maps intro price correctly") {
+                var receivedDictionary = self.storeProductDictionaryWith()
+
+                expect(receivedDictionary["introPrice"] as? SKProductDiscount?) == nil
+                let subscriptionPeriod = SKProductSubscriptionPeriod(numberOfUnits: 3, unit: .month)
+                let productDiscount = SKProductDiscount(price: 10.99,
+                                                        priceLocale: Locale.current,
+                                                        identifier: "product discount",
+                                                        subscriptionPeriod: subscriptionPeriod,
+                                                        numberOfPeriods: 3,
+                                                        paymentMode: SKProductDiscount.PaymentMode.payAsYouGo,
+                                                        type: .introductory)
+                receivedDictionary = self.storeProductDictionaryWith(introductoryPrice: productDiscount)
+                let receivedIntroPrice = try XCTUnwrap(receivedDictionary["introPrice"] as? [String:Any])
+                expect(receivedIntroPrice["price"] as? Decimal) == 10.99
+            }
+        }
+    }
+}
+
+public extension StoreProduct {
+    convenience init(priceLocale: Locale,
+                     localizedDescription: String,
+                     discounts: [SKProductDiscount],
+                     productIdentifier: String,
+                     introductoryDiscount: StoreProductDiscount?,
+                     price: Decimal,
+                     productType: ProductType,
+                     localizedTitle: String,
+                     subscriptionPeriod: SKProductSubscriptionPeriod?,
+                     isFamilyShareable: Bool,
+                     subscriptionGroupIdentifier: String,
+                     introductoryPrice: SKProductDiscount?
+    ) {
+        let skProduct = SKProduct(localizedDescription: localizedDescription,
+                                  localizedTitle: localizedTitle,
+                                  price: price, // Might need to convert to NSDecimalNumber
+                                  priceLocale: priceLocale,
+                                  productIdentifier: productIdentifier,
+                                  isDownloadable: true,
+                                  isFamilyShareable: isFamilyShareable,
+                                  downloadContentLengths: [1],
+                                  contentVersion: "1",
+                                  downloadContentVersion: "1",
+                                  subscriptionPeriod: subscriptionPeriod,
+                                  introductoryPrice: introductoryPrice,
+                                  subscriptionGroupIdentifier: subscriptionGroupIdentifier,
+                                  discounts: discounts)
+
+        self.init(sk1Product: skProduct)
```

Comment:
Not sure how this will be done for sk2Product. But I figured this is good for now. I think the only way is using StoreKit tests

Reply:
Correct